### PR TITLE
[msbuild] Improve altool task by logging execution errors

### PR DESCRIPTION
**Pull request by emaf**
The altool task was just logging the XML output produced by the tool execution but was not logging any build error neither failing in that case.

### CODE REVIEW

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs [hunk 1]
```diff
@@ -82,5 +96,28 @@ string GetFileTypeValue ()
...
+				if ((plist?.TryGetValue ("product-errors", out errors) == true)) {
+					foreach (var error in errors) {
+						var dict = error as PDictionary;
+						if (dict?.TryGetValue ("message", out message) == true) {
+							Log.LogError (ToolName, null, null, null, 0, 0, 0, 0, "{0}", message.Value);
+						}
+					}
+				}
+			} catch (Exception ex) {
+				Log.LogWarning ("Failed to parse altool output: {0}", ex.Message);
```
Comment (spouliot) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs]: Should you include `output` in the exception log ?
That way you can get a clue of what went wrong

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs [hunk 2]
```diff
@@ -82,5 +96,28 @@ string GetFileTypeValue ()
 				default: throw new NotSupportedException ($"Provided file type '{FileType}' is not supported by altool");
 			}
 		}
+
+		void LogErrorsFromOutput (string output)
+		{
+			try {
+				if (string.IsNullOrEmpty(output))
```
Comment (spouliot) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs]: minor: space before `(`

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs [hunk 3]
```diff
@@ -82,5 +96,28 @@ string GetFileTypeValue ()
...
+
+				var plist = PObject.FromString (output) as PDictionary;
+				var errors = PObject.Create (PObjectType.Array) as PArray;
+				var message = PObject.Create (PObjectType.String) as PString;
+
+				if ((plist?.TryGetValue ("product-errors", out errors) == true)) {
+					foreach (var error in errors) {
+						var dict = error as PDictionary;
+						if (dict?.TryGetValue ("message", out message) == true) {
+							Log.LogError (ToolName, null, null, null, 0, 0, 0, 0, "{0}", message.Value);
```
Comment (spouliot) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs]: not sure if the API (or some overloads) allows it but `"{0}"` seems to require a call to `String.Format`, can't you just provide `message.Value` ?

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs (hunk 1 above)
Comment (spouliot) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs]: also `$"Failed to parse altool output: {ex.Message}"` looks nicer :)

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs [hunk 4]
```diff
@@ -82,5 +96,28 @@ string GetFileTypeValue ()
...
+					}
+				}
+			} catch (Exception ex) {
+				Log.LogWarning ("Failed to parse altool output: {0}", ex.Message);
+			}
+		}
 	}
-}
\ No newline at end of file
+}
```
Comment (rolfbjarne) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs]: Unnecessary space change

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs (hunk 3 above)
Comment (rolfbjarne) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs]: @spouliot What if `message.Value` contains something that can be a format string? Say `"{0}"` for instance, then things might fail (depending on LogError's implementation).

Comment (spouliot) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs]: yeah, bring us back to _not sure if the API_ :)

Comment (emaf) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs]: yes, I don't really like it (using `"{0}"`) but it seems to be _safer_, and that's how most of the XI tasks log errors that come from tool outputs or text we don't control.

**Comment by rolfbjarne** [comment]
build

**Comment by rolfbjarne** [comment]
build

**Comment by emaf** [comment]
@rolfbjarne could you review the latest changes? Is there anything else needed to get this PR merged?

**Pull request merged**